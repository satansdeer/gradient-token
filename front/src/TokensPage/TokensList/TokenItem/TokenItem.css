.TokenItem {
  cursor: pointer;
  position: relative;
  animation: tokenItemIn .25s cubic-bezier(.06,.67,.37,.99) forwards
}

.TokenItem:hover > .TokenItem-image_wrapper {
  box-shadow: rgb(255, 255, 255) 0px 0px 0px 4px inset
}

.TokenItem-image_wrapper {
  background-color: #e7dcd6;
  border-radius: 10px;
  padding: 40px 20px;
  margin-bottom: 15px;
  box-shadow: rgb(255, 255, 255) 0px 0px 0px 0px inset;
  transition: box-shadow 0.2s;
}

.TokenItem-label {
  text-align: left;
}

@keyframes tokenItemIn {
  0%   { opacity: 0; transform: translateY(10px) }
  100% { opacity: 1; transform: translateY(0) }
}
