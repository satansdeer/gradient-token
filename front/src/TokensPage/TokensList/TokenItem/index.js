import React from "react";
import PropTypes from "prop-types";
import { PropTypes as MobxPropTypes } from "mobx-react";

import TokenImage from "components/TokenImage";
import "./TokenItem.css";

const TokenItem = ({ token, onClick }) => {
  return (
    <div className="TokenItem" onClick={onClick}>
      <div className="TokenItem-image_wrapper">
        <TokenImage outer={token[0]} inner={token[1]} />
      </div>
      <div className="TokenItem-label">{`${token[0]} â€“ ${token[1]}`}</div>
    </div>
  );
};

TokenItem.propTypes = {
  token: MobxPropTypes.arrayOrObservableArray,
  onClick: PropTypes.func
};

export default TokenItem;
