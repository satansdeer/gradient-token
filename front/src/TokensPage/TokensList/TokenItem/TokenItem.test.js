import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import TokenItem from "./index";

describe("<TokenItem />", () => {
  it("renders and matches the snapshot", () => {
    const component = renderer.create(<TokenItem token={["#000", "#fff"]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls the supplied onClick callback when clicked", () => {
    const token = ["#000", "#fff"];
    const mockOnClick = jest.fn();
    const component = shallow(
      <TokenItem token={token} onClick={mockOnClick} />
    );
    component.simulate("click");
    expect(mockOnClick).toHaveBeenCalled();
  });
});
