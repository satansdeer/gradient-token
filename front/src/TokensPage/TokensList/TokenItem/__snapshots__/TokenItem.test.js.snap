// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokenItem /> renders and matches the snapshot 1`] = `
<div
  className="TokenItem"
  onClick={undefined}
>
  <div
    className="TokenItem-image_wrapper"
  >
    <svg
      height={100}
      version="1.1"
      width={100}
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <radialGradient
          id="000fff"
        >
          <stop
            offset="0%"
            stopColor="#000"
          />
          <stop
            offset="100%"
            stopColor="#fff"
          />
        </radialGradient>
      </defs>
      <circle
        cx={50}
        cy={50}
        fill="url(#000fff)"
        r={46}
        stroke="#fff"
        strokeWidth="2"
      />
    </svg>
  </div>
  <div
    className="TokenItem-label"
  >
    #000 â€“ #fff
  </div>
</div>
`;
